File: gj.pl
male(Ivan, A)
male(Timur, C)
female(Rita, B)
female(Irina, C)

% Молодец?
greatjob(X):-male(X,A);female(X,A);male(X,B);female(X,B).
% Мужчина?
ismale(X):-greatjob(X),male(X,_). 

File: gj.py
class Student:
    def __init__(self, name, gender, grades):
        self.name = name
        self.gender = gender
        self.grades = grades
    def well_done():
        self.well_done = True

student1 = Student("Ivan", "male", "A")
student2 = Student("Timur", "male", "C")
student3 = Student("Rita", "female", "B")
student4 = Student("Irina", "female", "C")

for i in [student1, student2, student3, student4]:
    if i.grades == "A" or i.grades == "B":
        i.well_done()
        
for i in [student1, student2, student3, student4]:
    if student.well_done and student.gender == "male":
        print("Male")